StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentMethodType.swift

```diff
@@ -211,14 +211,23 @@ extension PaymentSheet {
                 }
             }
 
-            return recommendedPaymentMethodTypes.filter {
-                PaymentSheet.PaymentMethodType.supportsAdding(
-                    paymentMethod: $0,
+            return recommendedPaymentMethodTypes.filter { paymentMethodType in
+                let exclusionReason = PaymentSheet.PaymentMethodType.supportsAdding(
+                    paymentMethod: paymentMethodType,
                     configuration: configuration,
                     intent: intent,
                     supportedPaymentMethods: configuration.linkPaymentMethodsOnly
                         ? PaymentSheet.supportedLinkPaymentMethods : PaymentSheet.supportedPaymentMethods
                 )
+
+                if exclusionReason != .supported {
+                    // This payment method is being flitered out, log the reason/s why
+                    #if DEBUG
+                    print("[Stripe SDK]: \(paymentMethodType.displayName) is not being displayed because one or more requirements are not being met or this payment method is not supported by PaymentSheet. \(exclusionReason.description)\n")
+                    #endif
```

Comment:
Can you move this into a helper function so we don't pollute our codebase with `#if DEBUG` checks?, it would also make it easier to change in the future

Reply:
Agree it would be nice to re-use in the future but it's only being used in one place for now an a function that is 
```swift
func debugPrint(str) {
   #if DEBUG
   print(str)
   #endif
 }
```
seems a bit overkill for one use, and there really isn't a place for this function to live that makes sense now. I think it's something we should tackle when we add a logging framework.

Reply:
it could live in StripeCore -> Helpers, and available through @_spi. The function could add the "[Stripe SDK]" prefix as well for example.
I agree that since it is a single use is not strictly necessary, but I'd rather add it now while you are in there, but either way is okay.

Reply:
I'm leaning towards not adding it because I don't think we want people using a helper logging function, we want people using a logging framework (that doesn't exist). Creating a new file in StripeCore e.g. `Logger.swift` just for this  function will indicate to folks that this is the blessed way to do logging in our SDK when it is not. If we find a need for a helper like this I think we should definitely add a logging framework. 

https://paper.dropbox.com/doc/2021-10-11-Shared-Logging-APIs-Mobile-API-Review-R2HlpenOCskNdl37HWHY8

Reply:
that makes sense, let's skip this